Good-Bids/goodbids
# [#11] Main Header and Footer

# Summary
We are setting up the header and footer for the main GoodBids site. 
The header is pretty much a copy of the child theme, but with a few things removed. 
The footer is rebuilt using a custom block that is set up in our plugin. The custom block is hard coded with fake links, until we get #90 set up. 
Putting the bottom footer content into a block prevents it from being edited and allows us to lock the block which prevents it from being deleted or moved. I added a filter so only Super Admins are allowed to lock/unlock site blocks. 
Any edits to the food block will roll out to the rest of the network sites.

## Issues

* #11

## Testing Instructions

### Test on Main
- Make sure the Main site is using the new GoodBids Main theme. 
- Then go to Appearance -> Editor
- Then go to Patterns -> Footer and click the "pen" icon to edit the footer.  
- You may not see the new bottom footer as your instance may have been set up before the new footer block.
  - If you don't see the new footer go to add a new block and insert `Main Footer`.
  - Save out the footer.
- You should now see the new footer on the site.

### Test on Nonprofit
- Make sure the Main site is using the new GoodBids Main theme. 
- Set up a new site
  -  Network Admin -> Sites -> Add New Site
- View the site and make sure the nonprofit site has both footers (nonprofit and global)
- Go to the dashboard of the new site and click on Users.
- Click on 'Switch To' to switch to that user on the new site.
- Then go to Appearance -> Editor
- Then go to Patterns -> Footer and click the "pen" icon to edit the footer.  
- Make sure you can not unlock/edit the Main Footer block.


## Screenshots

|  Super Admins | All other users |
|--------|--------|
| ![Screenshot 2023-12-20 at 6 21 00 PM](https://github.com/Good-Bids/goodbids/assets/91974372/eb9150c6-20c8-42e1-a2cb-0ba1426c45ac) | ![Screenshot 2023-12-20 at 6 21 36 PM](https://github.com/Good-Bids/goodbids/assets/91974372/aeff2463-ec4b-416f-b223-2448e9415fb3)|


### Main site
![Screenshot 2023-12-20 at 5 46 07 PM](https://github.com/Good-Bids/goodbids/assets/91974372/9a5040a7-f626-4916-a76e-d1fa96b00608)

### Child site
![Screenshot 2023-12-20 at 5 46 58 PM](https://github.com/Good-Bids/goodbids/assets/91974372/508a8682-28aa-4d30-aef2-3215d7d6cb56)
